workspace_tools/export/gcc_arm_saml21j18a.tmpl

```diff
@@ -26,7 +26,7 @@ CC_SYMBOLS = {% for s in symbols %}-D{{s}} {% endfor %}
 
 LD_FLAGS = $(CPU) -Wl,--gc-sections --specs=nano.specs -u _printf_float -u _scanf_float -Wl,--wrap,main
 LD_FLAGS += -Wl,-Map=$(PROJECT).map,--cref
-LD_SYS_LIBS = -lstdc++ -lsupc++ -lm -lc -lgcc -lnosys
+LD_SYS_LIBS = -lstdc++ -lsupc++ -lm -lgcc -Wl,--start-group -lc -lc -lnosys -Wl,--end-group 
```

Comment:
Atmel studio uses arm-none-eabi-g++ for linking. Hence made the makefile template to use the same. But using arm-none-eabi-g++ causes below linker error 
```		
c:/program files (x86)/atmel/atmel toolchain/arm gcc/native/4.8.1443/arm-gnu-toolchain/bin/../lib/gcc/arm-none-eabi/4.8.4/../../../../arm-none-eabi/lib/armv6-m\libc_s.a(lib_a-signalr.o): In function `_kill_r':
D:\Workspace\mbed\drivers-atmel\build\export\MBED_A3_atmelstudio_SAMR21G18A\MBED_A3\Debug\signalr.c(1,1): error: undefined reference to `_kill'
		c:/program files (x86)/atmel/atmel toolchain/arm gcc/native/4.8.1443/arm-gnu-toolchain/bin/../lib/gcc/arm-none-eabi/4.8.4/../../../../arm-none-eabi/lib/armv6-m\libc_s.a(lib_a-signalr.o): In function `_getpid_r':
D:\Workspace\mbed\drivers-atmel\build\export\MBED_A3_atmelstudio_SAMR21G18A\MBED_A3\Debug\signalr.c(1,1): error: undefined reference to `_getpid'
```
With some googling figured out that libnosys also has these symbols. It seems to be caused by recursive symbol dependence. Using "-lc -lc -lnosys" within start and end-group resolves this issue.